eclass/tests/flag-o-matic.sh

```diff
@@ -146,4 +146,74 @@ out=$(CC=clang test-flags-CC -finline-limit=1200)
 ftend
 fi
 
+clear-env() {
+	unset $(all-flag-vars)
+}
+
+fom-tbegin() {
+	clear-env
```

Comment:
I've had this idea as well, unfortunately after I've written all of it and haven't considered it to be worth the effort to change it.